### PR TITLE
Hide unused models.

### PR DESCRIPTION
**Pull request by katmsft**
# Pull Request Checklist

Please leverage this checklist as a reminder to address commonly occurring feedback when submitting a pull request to make sure your PR can be reviewed quickly:

See the detailed list in the [contributing guide](https://github.com/Azure/azure-sdk-for-cpp/blob/master/CONTRIBUTING.md#pull-requests).

- [x] [C++ Guidelines](https://azure.github.io/azure-sdk/cpp_introduction.html)
- [x] Doxygen docs
- [x] Unit tests
- [x] No unwanted commits/changes
- [x] Descriptive title/description
  - [x] PR is single purpose
  - [x] Related issue listed
- [x] Comments in source
- [x] No typos
- [x] Update changelog
- [x] Not work-in-progress
- [x] External references or docs updated
- [x] Self review of PR done
- [x] Any breaking changes?

### CODE REVIEW

**Review by Jinming-Hu — APPROVED** [review]

File: sdk/storage/azure-storage-files-shares/CHANGELOG.md [hunk 1]
```diff
@@ -25,6 +25,12 @@
 - Removed `RequestId` from the return types.
 - Changed `BodyStream` parameter of `UploadRange` function from pointer to reference.
 - Removed `PreviousShareSnapshot` from `GetShareFileRangeListOptions`, use `ShareFileClient::GetRangeListDiff` instead.
+- Removed `PathRenameMode`.
+- Removed `PathGetPropertiesAction`.
+- Removed `PathSetAccessControlRecursiveMode`.
+- Removed `FileSystemResourceType`.
+- Removed `FileSystemResourceType`.
+- Removed `IsAccessTierInferred` and `AccessTierChangedOn` from `PathProperties`.
```
Comment (Jinming-Hu) [inline sdk/storage/azure-storage-files-shares/CHANGELOG.md:33]: Merge into one entry?

Comment (ahsonkhan) [inline sdk/storage/azure-storage-files-shares/CHANGELOG.md:33]: I am curious, how come we can remove these from `PathProperties` (it's fine that we are, just wondering)? Why did we have them to begin with?

Comment (katmsft) [reply sdk/storage/azure-storage-files-shares/CHANGELOG.md:33]: Access Tier related things are not available for datalake service. Jinming and I had a discussion and decide that we should remove them. However this API calls blob API, which will return them sometimes. That's why we originally have those fields.

**Pull request merged**